### PR TITLE
Minor improvements to PolymerProposalEngine

### PR DESCRIPTION
**Pull request by zhang-ivy**
## Description

Here, I am adding the following improvements to `PolymerProposalEngine`:
1. For protein mutations, do not map sidechain atoms (CB atoms and beyond)
2. Speed up the execution of `construct_atom_map`
3. Handle ring amino acids

## Motivation and context

Changes:
1. We don't want to map sidechain atoms for protein mutations because of the sidechain rotamer sampling problem. We can use REST to enhance sampling if we do not map sidechain atoms.
2. We can speed up the execution of `construct_atom_map` by removing one of the nested for loops.
3. We need to set `allow_ring_breaking=False` for ring amino acids.

## How has this been tested?

As long as the tests in `test_topology_proposal.py` run ok, we should be fine.

## Change log

```

```

### CODE REVIEW

File: perses/rjmc/topology_proposal.py [hunk 1]
```diff
@@ -1137,6 +1138,8 @@ def _construct_atom_map(self,
             key : int (index) , value : str (three letter name of proposed residue)
         new_topology : simtk.openmm.app.Topology
             topology of new system
+        ignore_sidechain_atoms : bool, default False
+            whether to ignore sidechain atoms in the atom map
```
Comment (dominicrufa) [inline perses/rjmc/topology_proposal.py:1142]: this includes CB atoms and beyond?

File: perses/rjmc/topology_proposal.py [hunk 2]
```diff
@@ -1334,14 +1339,18 @@ def _construct_atom_map(self,
...
-            _logger.debug(f"\t\t\tno 'CB' found!!!.  removing local atom map stereo sidechain...")
-            sidechain_fixed_map = {}
+        #make sure that CB is mapped; otherwise the residue will not be contiguous
+        #found_CB = False
+        #if any(item[0] == 'CB' and item[1] == 'CB' for item in mapped_names):
+        #    found_CB = True
+	#
+        #if not found_CB:
+        #    _logger.debug(f"\t\t\tno 'CB' found!!!.  removing local atom map stereo sidechain...")
+        #    sidechain_fixed_map = {}
```
Comment (dominicrufa) [inline perses/rjmc/topology_proposal.py:1353]: can safely delete this bit, i think

**Review by dominicrufa — APPROVED** [review]

File: perses/rjmc/topology_proposal.py (hunk 1 above)
Comment (zhang-ivy) [reply perses/rjmc/topology_proposal.py:1142]: yes. added this clarification to the docstring

File: perses/rjmc/topology_proposal.py (hunk 2 above)
Comment (zhang-ivy) [reply perses/rjmc/topology_proposal.py:1353]: ok

**Review by mikemhenry — APPROVED** [review]

**Pull request merged**